source/extensions/filters/udp/dns_filter/dns_filter.h

```diff
@@ -42,24 +42,44 @@ using DnsVirtualDomainConfig = absl::flat_hash_map<std::string, AddressConstPtrV
 /**
  * DnsFilter configuration class abstracting access to data necessary for the filter's operation
  */
-class DnsFilterEnvoyConfig {
+class DnsFilterEnvoyConfig : public Logger::Loggable<Logger::Id::filter> {
 public:
   DnsFilterEnvoyConfig(
       Server::Configuration::ListenerFactoryContext& context,
       const envoy::extensions::filters::udp::dns_filter::v3alpha::DnsFilterConfig& config);
 
+  DnsFilterStats& stats() const { return stats_; }
+  const DnsVirtualDomainConfig& domains() const { return virtual_domains_; }
+  const std::vector<Matchers::StringMatcherPtr>& knownSuffixes() const { return known_suffixes_; }
+  const absl::flat_hash_map<std::string, std::chrono::seconds>& domainTtl() const {
+    return domain_ttl_;
+  }
+  const AddressConstPtrVec& resolvers() const { return resolvers_; }
+  bool forwardQueries() const { return forward_queries_; }
+  const std::chrono::milliseconds resolverTimeout() const { return resolver_timeout_; }
+  Upstream::ClusterManager& clusterManager() const { return cluster_manager_; }
+  uint64_t retryCount() const { return retry_count_; }
+
 private:
   static DnsFilterStats generateStats(const std::string& stat_prefix, Stats::Scope& scope) {
     const auto final_prefix = absl::StrCat("dns_filter.", stat_prefix);
     return {ALL_DNS_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
   }
 
+  bool loadServerConfig(const envoy::extensions::filters::udp::dns_filter::v3alpha::
+                            DnsFilterConfig::ServerContextConfig& config,
+                        envoy::data::dns::v3::DnsTable& table);
+
   Stats::Scope& root_scope_;
+  Upstream::ClusterManager& cluster_manager_;
+  Api::Api& api_;
+
   mutable DnsFilterStats stats_;
   DnsVirtualDomainConfig virtual_domains_;
   std::vector<Matchers::StringMatcherPtr> known_suffixes_;
   absl::flat_hash_map<std::string, std::chrono::seconds> domain_ttl_;
   bool forward_queries_;
+  uint64_t retry_count_;
```

Comment:
can this be a const?

Reply:
In this context it's populated after the class is initialized when we read from the config, so no.